storage/gcloud/aio/storage/storage.py

```diff
@@ -401,6 +401,25 @@ async def _download(self, bucket: str, object_name: str, *,
 
         return data
 
+    async def _download_stream(self, bucket: str, object_name: str, *,
+                        params: Optional[Dict[str, str]] = None,
+                        headers: Optional[Dict[str, str]] = None,
+                        timeout: int = 10,
+                        session: Optional[Session] = None) -> StreamResponse:
+        # https://cloud.google.com/storage/docs/json_api/#encoding
+        encoded_object_name = quote(object_name, safe='')
+        url = f'{API_ROOT}/{bucket}/o/{encoded_object_name}'
+        headers = headers or {}
+        headers.update(await self._headers())
+
+        s = AioSession(session) if session else self.session
+        response = await s.get(url, headers=headers, params=params or {},
+                               timeout=timeout, stream=True)
+
+        if BUILD_GCLOUD_REST:
+            return response.iter_content
+        return response.content
+
```

Comment:
@TheKevJames WDYT about this implementation?

As you suggested, I added to the conditional types at the top, creating a `StreamResponse` type variable. After wading through `requests.Response` for a while, I concur -- `iter_content` is the way to go, and I think the most similar thing to return is the `iter_content` generator itself. There are no type hints in requests, so I just went with the Generator type there.

Reply:
I think this conditional works fine -- often we would write this as eg.
```python
try:
    return response.iter_content
except AttributeError:
    return response.content
```
but I think this `if` check is clearer anyway. LGTM as-is :+1: